packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -219,8 +219,8 @@ export class Outbox {
 			return this.params.groupingManager.groupBatch(batch);
 		}
 
-		const compressedBatch = this.params.groupingManager.groupBatch(
-			this.params.compressor.compressBatch(batch),
+		const compressedBatch = this.params.compressor.compressBatch(
+			this.params.groupingManager.groupBatch(batch),
```

Comment:
I think it does what it should, but are there any back-compat implications here? I.e. we already have compression in production.
I believe we are fine (as long as we roll these changes slowly), but worth ensuring that we test it thoroughly. I think we need to add a test coverage for already in production format. Normally, we did so by adding new case into snapshot tests, but we have not done it for a while, and snapshot tests are built with assumption that ops alone is sufficient to load container - something that is not true anymore, so it's likely hard to add new test cases, but it feels like we need to fix it for features like that to be able to safely move forward and have test coverage.
Might be worth adding a separate task to follow up, as we need to ensure we do not break old formats!

Reply:
The back-compat message is probably more appropriate for changes in next file :)

Reply:
I think compat concerns would only be for reading previous order when grouped batching and compression are applied. Old version will try to ungroup first whereas now we decompress first.

This can be fixed by making ungrouping calls before and after the decompress calls, but the loop will get harder to read.

Reply:
I'm becoming more and more paranoid about these types of issues so my 2c is that we should probably try to cover all our bases and go ahead and do a double take wrt to ungrouping (like we do with chunking) even if it reads harder (we can probably do some refactoring to make it prettier).